src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/google/rpc/error_details.proto

```diff
@@ -0,0 +1,285 @@
+    // which terms of service is being referenced.
+    string subject = 2;
+
+    // A description of how the precondition failed. Developers can use this
+    // description to understand how to fix the failure.
+    //
+    // For example: "Terms of service not accepted".
+    string description = 3;
+  }
+
+  // Describes all precondition violations.
+  repeated Violation violations = 1;
+}
+
+// Describes violations in a client request. This error type focuses on the
+// syntactic aspects of the request.
+message BadRequest {
+  // A message type used to describe a single bad request field.
+  message FieldViolation {
+    // A path that leads to a field in the request body. The value will be a
+    // sequence of dot-separated identifiers that identify a protocol buffer
+    // field.
+    //
+    // Consider the following:
+    //
+    //     message CreateContactRequest {
+    //       message EmailAddress {
+    //         enum Type {
+    //           TYPE_UNSPECIFIED = 0;
+    //           HOME = 1;
+    //           WORK = 2;
+    //         }
+    //
+    //         optional string email = 1;
+    //         repeated EmailType type = 2;
+    //       }
+    //
+    //       string full_name = 1;
+    //       repeated EmailAddress email_addresses = 2;
+    //     }
+    //
+    // In this example, in proto `field` could take one of the following values:
+    //
+    // * `full_name` for a violation in the `full_name` value
+    // * `email_addresses[1].email` for a violation in the `email` field of the
+    //   first `email_addresses` message
+    // * `email_addresses[3].type[2]` for a violation in the second `type`
+    //   value in the third `email_addresses` message.
+    //
+    // In JSON, the same values are represented as:
+    //
+    // * `fullName` for a violation in the `fullName` value
+    // * `emailAddresses[1].email` for a violation in the `email` field of the
+    //   first `emailAddresses` message
+    // * `emailAddresses[3].type[2]` for a violation in the second `type`
+    //   value in the third `emailAddresses` message.
+    string field = 1;
+
+    // A description of why the request element is bad.
+    string description = 2;
+  }
+
+  // Describes all violations in a client request.
+  repeated FieldViolation field_violations = 1;
+}
+
+// Contains metadata about the request that clients can attach when filing a bug
+// or providing other forms of feedback.
+message RequestInfo {
+  // An opaque string that should only be interpreted by the service generating
+  // it. For example, it can be used to identify requests in the service's logs.
+  string request_id = 1;
+
+  // Any data that was used to serve this request. For example, an encrypted
+  // stack trace that can be sent back to the service provider for debugging.
+  string serving_data = 2;
+}
+
+// Describes the resource that is being accessed.
+message ResourceInfo {
```

Comment:
Could we remove the unused `message` definitions from this file?

Reply:
Sure, often when I forklift stuff I tend to keep it all in tact for ease of maintenance, but likely we'll never need to touch these files again.